scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
```

Comment:
All files should start with `from __future__ import division, absolute_import, print_function` stanza. For most purposes, this is enough to guarantee Python 3 compatibility.
I guess the only way to check for compatibility is to run the code also on Python 3.
(The continuous integration test runner runs it also on Py3)